### ISSUE TITLE
Proposal: Combined conditions [like if (a <= b <= c)]

### ISSUE THREAD

**Issue by pekspro** [dotnet/csharplang#1194]
If I need to check if a value, y, is between x and z I could write a statement like this:

    if (x <= y && y <= z)

It's fine, but it would be even better if I just needed to write:

    if (x <= y <= z)

Of course, I want to use the operators as well and using more conditions, for instance:

    if (a == b == c < d)

to check if a, b and c has the same value and are less the d. This should be executed like:

* First testing if a == b is true.
* It if this, then test if b == c is true
* It if this, then test if c < d is true

Just like:

    if (a == b && b == c && c < d)

Checking if a point is inside a rectangle could be expressed in this elegant way:

    if( rect.Left < point.X < rect.Right && rect.Bottom < point.Y < rect.Top )

If I have an expression, like tree(3) + 1, that's take time to execute and I want to test if the result is between a and b writing:

    if (a == Tree(3) + 1 && Tree(3) + 1 == b)

would be inefficient (and ugly code). Instead I should introduce a local variable that holds the value of the expression. But if I instead could write:

    if (a <= Tree(3) + 1 <= b)

Then the compiler could make sure that the expression is only executed once.

In summary would like to use this to write simple if-statements and avoid introducing local variables for heavy calculations.  I'm confident that there are tons of special cases I haven't thought about. I just wanted to share my idea. I hope it's clear :-).

**Comment by cpplearner** [comment]
#1175
#1128
#217
#847
#420

**Comment by Joe4evr** [comment]
Sounds like part of this is a duplicate of ~~#847~~ all the issues @cpplearner just posted. 🤣 

The rest can be done once more pattern matching features are in. For example, your checking of a point inside a rectangle could turn into something like:
```cs
if (point.X is in rect.Left..rect.Right && point.Y is in rect.Top..rect.Bottom)
```
(Should be noted that `is in` is *not* a confirmed pattern syntax yet. Also, follow the official naming conventions and name your public members in PascalCase.)

**Comment by pekspro** [comment]
Nice to see other people have the same thoughts :-) I actually tried to find similar posts but failed to find any.

Either way, I'm closing this since this is a duplicate.
